core-client/src/lib.rs

```diff
@@ -1581,6 +1590,150 @@ async fn do_restore_from_backup(
     Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+pub async fn do_reshare(
+    internal_client: &mut Client,
+    core_endpoints: &HashMap<u32, CoreServiceEndpointClient<Channel>>,
+    rng: &mut AesRng,
+    cmd_conf: &CmdConfig,
+    cc_conf: &CoreClientConfig,
+    destination_prefix: &Path,
+    kms_addrs: &[alloy_primitives::Address],
+    num_parties: usize,
+    param: FheParameter,
+    key_id: RequestId,
+    preproc_id: RequestId,
+    max_retries: usize,
```

Comment:
Is this different from `cmd_conf.max_iter;`? I guess it is, since this is how many times we might want to reexecute a failed protocol. 
Could you add a comment about this?